### DISCUSSION TITLE
Schneider S520619 Thermostat

### DISCUSSION THREAD

**Discussion by ChrisLou14** [Koenkk/zigbee2mqtt#24711] [Devices]
Hello, I have buy a schneider thermostat but unfortunately it was not supported by Z2m. 
So I try to follow the instructions to include the device and I have many difficulties .
The device was include to my z2m and I can see it on the z2m interface.
I have e few data read (occupancy, link quality, temperature) but I want to write an read occupiedHeatingSetpoint (endpoint 1 binding havthermostat.
I can read and write this item from the console dev but I can't collect it by Z2m.
 Can you help me ?

(sorry for my bad English I'm french)

**Comment by leperroy** [comment]
I'm using as well this device. Will be happy if Z2M would support it.

**Comment by pcurt** [comment]
Hello, I'm on it !

**Comment by pcurt** [comment]
I have open a PR https://github.com/Koenkk/zigbee-herdsman-converters/pull/8452 that adds the support.
You can already use it using an external converter with this content

```
const {} = require('zigbee-herdsman-converters/lib/modernExtend');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const {} = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;


const definition = {
    zigbeeModel: ['S520619'],
    model: 'S520619',
    vendor: 'Schneider Electric',
    description: 'Wiser Odace Smart thermostat',
        fromZigbee: [fz.stelpro_thermostat, fz.metering, fz.schneider_pilot_mode, fz.wiser_device_info, fz.hvac_user_interface, fz.temperature, fz.occupancy],
        toZigbee: [
            tz.thermostat_occupied_heating_setpoint,
            tz.thermostat_system_mode,
            tz.thermostat_local_temperature,
            tz.thermostat_control_sequence_of_operation,
            tz.schneider_pilot_mode,
            tz.schneider_thermostat_keypad_lockout,
            tz.thermostat_temperature_display_mode,
        ],
        exposes: [
            e.binary('keypad_lockout', ea.STATE_SET, 'lock1', 'unlock').withDescription('Enables/disables physical input on the device'),
            e.enum('schneider_pilot_mode', ea.ALL, ['contactor', 'pilot']).withDescription('Controls piloting mode'),
            e
                .enum('temperature_display_mode', ea.ALL, ['celsius', 'fahrenheit'])
                .withDescription('The temperature format displayed on the thermostat screen'),
            e
                .climate()
                .withSetpoint('occupied_heating_setpoint', 4, 30, 0.5)
                .withLocalTemperature()
                .withSystemMode(['off', 'heat', 'cool'])
                .withPiHeatingDemand(),
            e.temperature(),
            e.occupancy(),
        ],
        configure: async (device, coordinatorEndpoint) => {
            const endpoint1 = device.getEndpoint(1);
            const endpoint2 = device.getEndpoint(2);
            const endpoint4 = device.getEndpoint(4);
            await reporting.bind(endpoint1, coordinatorEndpoint, ['hvacThermostat']);
            await reporting.thermostatPIHeatingDemand(endpoint1);
            await reporting.thermostatOccupiedHeatingSetpoint(endpoint1);
            await reporting.temperature(endpoint2);
            await endpoint1.read('hvacUserInterfaceCfg', ['keypadLockout', 'tempDisplayMode']);
            await reporting.bind(endpoint4, coordinatorEndpoint, ['msOccupancySensing']);
            await reporting.thermostatOccupancy(endpoint4);
        },
        options: [exposes.options.measurement_poll_interval()],
        onEvent: async (type, data, device, options) => {
            const endpoint = device.getEndpoint(1);
            console.log("PCURT")
            const poll = async () => {
                await endpoint.read('hvacThermostat', ['occupiedHeatingSetpoint']);
            };
            utils.onEventPoll(type, data, device, options, 'measurement', 20, poll);
         },
};

module.exports = definition;
```

**Comment by pcurt** [comment]
FYI : my PR has been merged and is integrated in dev branch 
https://github.com/Koenkk/zigbee2mqtt/commit/6905c10a05479bef762138b18b4d6fd14e992c25

**Comment by eutampieri** [comment]
Is there a way to check if the thermostat has commanded heating/cooling? Does this need to be done by reading the pi_heating_demand? HA reports the device as always running...

**Reply by pcurt** [reply]
Yes I have checked with my thermostat. The behavior is quite strange, when a heating is asked `pi_heating_demand` is set to `39` when it is off it reports `0`
But I think I can add the  `pi_heating_demand` report to detect if the thermostat is heating ?

**Reply by svhelge** [reply]
> when a heating is asked `pi_heating_demand` is set to `39`

 This can be resolved by adding `meta: {thermostat: {dontMapPIHeatingDemand: true}}` to the device definition, similar to [this example](https://github.com/Koenkk/zigbee-herdsman-converters/blob/e5bb673eaaca70058c0aebf75631e23a2741ed9b/src/devices/schneider_electric.ts#L2798). The main point to verify is whether the device uses $100$ or $255$ as the maximum value for the piHeatingDemand attribute. When the device reports 39 it usually means that max is 100, not 255 which is Zigbee default (100/255 = 39.2).

**Comment by koanya** [comment]
Hi @pcurt . It is possible to add the occupied_cooling_setpoint ? Currently, we can only set the heating setpoint so in cool mode we can't set the setpoint through zigbee.
Thanks

**Reply by pcurt** [reply]
Hi @koanya,
Yes I will add it.

**Reply by pcurt** [reply]
@koanya it has been merged in last version 2.6.1
https://github.com/Koenkk/zigbee2mqtt/releases/tag/2.6.1
<img width="572" height="599" alt="Capture d’écran 2025-09-01 à 21 23 37" src="https://github.com/user-attachments/assets/76754135-3d4d-46f2-9715-790331e6e96a" />

**Reply by koanya** [reply]
Thank you @pcurt 👍

**Comment by leperroy** [comment]
This thermostat can be configured through the Wiser box to keep the temperature display always visible and adjust the brightness, but this functionality doesn't work when using Home Assistant. Is there a way to implement this in Home Assistant, or is there a workaround available?

**Reply by pcurt** [reply]
I have checked all the available endpoints and I don't know how to do it. We need someone with the Wiser box that can sniff the Zigbee network to see what is endpoint, the cluster and the attribute used to control the display.

**Comment by moriceh** [comment]
Hi !
The Schneider manual says that the cycle time for the relay can be edited within the Wiser app (with possible values being 5, 10, 20 and 30 minutes), do you think this could be implemented in z2m ?

Thanks !

**Reply by pcurt** [reply]
Hi, same response as previous request. 
To do it we need the wiser app and wiser box to reverse engineering the communication between the thermostat and the wiser gateway.

**Comment by svhelge** [comment]
It may be worth investigating whether S520619 shares attributes/clusters with the EKO07259. Detailed cluster definitions for the EKO07259 are available [here](https://github.com/user-attachments/files/16744437/RTC-6754012INT-cFMT_16A_F_Zigbee-29.pdf). According to the manual, the EKO07259 is functionally identical to the Exxact 16A (WDE002497) and the Renova 16A (WDE011680). 

I suspect the easiest way to test this theory is to spin up a quick modernExtend based converter and see how the device responds.
